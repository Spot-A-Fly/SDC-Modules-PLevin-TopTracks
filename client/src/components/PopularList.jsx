import React from 'react';
import PropTypes from 'prop-types';
import PopularTrack from './PopularTrack.jsx';

function PopularList({ tracks }) {
  return (
    <div>
      {tracks.map(track => <PopularTrack key={track._id} track={track} />)}
    </div>
  );
}

PopularList.propTypes = {
  tracks: PropTypes.isRequired,
};

export default PopularList;
