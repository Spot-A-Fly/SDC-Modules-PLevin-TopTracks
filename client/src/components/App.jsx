import React, { Component } from 'react';
import fetch from 'node-fetch';
import PopularList from './PopularList.jsx';

class App extends Component {
  constructor() {
    super();
    this.state = {
      tracks: []
    };
  }

  componentDidMount() {
    this.getTopTracks();
  }

  getTopTracks() {
    fetch('http://localhost:3003/data/toptracks')
      .then(results => results.json())
      .then((tracks) => {
        this.setState({ tracks });
      })
      .catch(console.log);
  }

  render() {
    const { tracks } = this.state;
    if (!tracks) return null;
   
    return (
      <div id="main" data-testid="popular-main">
        <div id="left" />
        <div id="content">
          <h1 id="header">Popular</h1>
          <PopularList data-testid="popular-list" tracks={tracks} />
        </div>
      </div>
    );
  }
}

export default App;
