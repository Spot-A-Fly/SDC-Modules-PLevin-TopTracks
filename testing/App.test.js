import React from 'react';
import { render, cleanup, waitForElement } from 'react-testing-library';
import App from '../client/src/components/App.jsx';
import PopularList from '../client/src/components/PopularList.jsx';

global.fetch = require('jest-fetch-mock');

afterEach(() => {
  cleanup();
  console.error.mockClear();
});

console.error = jest.fn();

const track = [
  {
    name: 'test_name',
    artist: 'test_artist',
    image: 'test_image',
    playCount: 100,
    length: 'test_length',
  }
];

test('<App />', async () => {
  fetch.mockResponseOnce(JSON.stringify(track));
  
  const { getByTestId } = render(<PopularList tracks={track} />);
  const { container } = render(<App />);
  await waitForElement(() => getByTestId('popular-list'));
  
  expect(getByTestId('popular-list').firstChild.textContent).toContain('test_name');
  expect(getByTestId('popular-list').firstChild.textContent).toContain('test_length');
  expect(getByTestId('track-image').src).toBe('http://localhost:3003/test_image');
  expect(container.firstChild).toMatchSnapshot();
});
