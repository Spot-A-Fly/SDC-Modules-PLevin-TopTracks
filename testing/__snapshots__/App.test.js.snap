// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> 1`] = `
<div
  data-testid="popular-main"
  id="main"
>
  <div
    id="left"
  />
  <div
    id="content"
  >
    <h1
      id="header"
    >
      Popular
    </h1>
    <div
      data-testid="popular-list"
    />
  </div>
</div>
`;
